src/commands/createFunctionApp/FunctionAppHostingPlanStep.ts

```diff
@@ -27,10 +27,13 @@ export class FunctionAppHostingPlanStep extends AzureWizardPromptStep<IFunctionA
     }
 
     public shouldPrompt(context: IFunctionAppWizardContext): boolean {
-        return context.useConsumptionPlan === undefined || context.dockerfilePath !== undefined;
+        return context.useConsumptionPlan === undefined && context.dockerfilePath === undefined;
     }
 
     public async getSubWizard(_context: IFunctionAppWizardContext): Promise<IWizardOptions<IFunctionAppWizardContext> | undefined> {
+        if (_context.dockerfilePath) {
```

Comment:
nit: Remove the _ from context. That's used by TypeScript to indicate an unused parameter, but since you're using it now, it doesn't make much sense.